File: NuixClient/Scripts/CreateCase.py
import sys, getopt



def main(argv):
   path = ''
   name = ''
   description = ''
   investigator = ''
   try:
      opts, args = getopt.getopt(argv,"hi:o:",["path=","name=","description=","investigator="])
   except getopt.GetoptError:
      print 'test.py -i <inputfile> -o <outputfile>'
      sys.exit(2)
   for opt, arg in opts:
      if opt == '-h':
         print 'test.py -i <inputfile> -o <outputfile>'
         sys.exit()
      elif opt in ("-p", "--path"):
         inputfile = path
      elif opt in ("-n", "--name"):
         name = arg
      elif opt in ("-d", "--description"):
         description = arg
      elif opt in ("-i", "--investigator"):
         investigator = arg
       
         
   print "Creating Case"


    theCase = utilities.caseFactory.create(path, {
                                             'name' : name,
                                             'description' : description,
                                             'investigator' : investigator})

    print "Case Created"

    theCase.close()

if __name__ == "__main__":
   main(sys.argv[1:])

File: NuixClient/Scripts/CreateCase.rb
require 'optparse'

hash_options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: your_app [options]"
  opts.on('-p [ARG]', '--path [ARG]', "Path") do |v|
    hash_options[:pathArg] = v
  end
  opts.on('-n [ARG]', '--name [ARG]', "Specify the Name") do |v|
    hash_options[:nameArg] = v
  end
  opts.on('-d [ARG]', '--description [ARG]', "Specify the Description") do |v|
    hash_options[:descriptionArg] = v
  end
  opts.on('-i [ARG]', '--investigator [ARG]', "Specify the Investigator") do |v|
    hash_options[:investigatorArg] = v
  end
  opts.on('--version', 'Display the version') do 
    puts "VERSION"
    exit
  end
  opts.on('-h', '--help', 'Display this help') do 
    puts opts
    exit
  end
end.parse!

requiredArguments = [:pathArg, :nameArg, :descriptionArg, :investigatorArg]

unless requiredArguments.all? {|a| hash_options[a] != nil}
    puts "Missing arguments #{(requiredArguments.select {|a| hash_options[a] == nil}).to_s}"


else
    puts "Creating Case"
    
        the_case = utilities.case_factory.create(hash_options[:pathArg],
            :name => hash_options[:nameArg],
            :description => hash_options[:descriptionArg],
            :investigator => hash_options[:investigatorArg])
    puts "Case Created"
        the_case.close
    puts "Case Closed"
    
end




